ensure_tiller_installed.go

```diff
@@ -261,9 +261,7 @@ func (c *Client) EnsureTillerInstalledWithValues(ctx context.Context, values []s
 
 		o := func() error {
 			t, err := c.newTunnel()
-			if !installTiller && IsTillerNotFound(err) {
-				return backoff.Permanent(microerror.Mask(err))
```

Comment:
Sometimes when the updated tiller status is `Pending` only, it would break the retry loop and e2etest would fail. It's worth to retry in this case rather than simply returning.

Reply:
We need to be careful when changing this code and I'd prefer not to make major changes. 

We will likely need to upgrade to helm 2.15.0 but after that we may be able to move to Helm 3. The real fix here is to remove tiller. 😄

Reply:
I saw sometimes it also failed at `tooManyResultsError` since multiple tiller pods would run at the same time. 
https://github.com/giantswarm/helmclient/blob/2734fbbea648359692b6a6d3e8704026787cb053/helmclient.go#L738-L740

Do you think make sense to change like below, 
```golang
			if IsTillerNotFound(err) || IsTooManyResults(err) {
				return microerror.Mask(err)
			} else if err != nil {
				return backoff.Permanent(microerror.Mask(err))
			}
```

Reply:
However I'm not against this change if it is well tested. 

Removing the permanent backoff may make sense but IIRC max wait is 1 minute. So we need to make sure this doesn't slow down the tiller resources in the operators.

Reply:
@rossf7 I compared the time amount that circleCI took a week ago for `helmclient`, and it looks like doesn't have much difference. 
Moreover, those `if` statement would only executed when it's upgrading, so I think we are safe from removing `backoff.Permanent`

Reply:
@tomahawk28  My concern is how it behaves when vendored in app-operator. 

Please also test it manually so we're sure tiller can still be installed in tenant clusters OK.

Reply:
If it can recovers after `tooManyResultsError`, then fine.
I agree with ross here this has to be well tested.